drivers/dma/dma_silabs_siwx917.c

```diff
@@ -90,11 +98,154 @@ static inline int siwx917_dma_addr_adjustment(uint32_t adjustment)
 	}
 }
 
-static int dma_channel_config(const struct device *dev, RSI_UDMA_HANDLE_T udma_handle,
-			      uint32_t channel, struct dma_config *config,
-			      UDMA_Channel_Info *channel_info)
+/* Sets up the scatter-gather descriptor table for a DMA transfer */
+static int siwx917_sg_fill_desc(RSI_UDMA_DESC_T *descs, const struct dma_config *config)
+{
+	const struct dma_block_config *block_addr = config->head_block;
+	volatile RSI_UDMA_CHA_CONFIG_DATA_T *cfg;
```

Comment:
"volatile is harmful". You can get the same result with accessors (`sys_write32()`, `sys_read32()`, etc...).